### DISCUSSION TITLE
Prevent certain adapters from detecting tests in the wrong language?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [nvim-neotest/neotest#256] [Q&A] [answered]
In one project I have both Python and Javascript/Typescript code. Here, the [neotest-mocha](https://github.com/adrigzr/neotest-mocha) adapter is wrongly detecting the Python tests and will attempt to run them. But of course I want the neotest-python adapter to detect those and run those tests, not neotest-mocha.

I opened up an issue with neotest-mocha [here](https://github.com/adrigzr/neotest-mocha/issues/7) about it, as my understanding is that it is each adapter's responsibility to detect the appropriate tests, e.g. based on file type/language (as mentioned [here](https://github.com/nvim-neotest/neotest#writing-adapters)).

But I was also thinking maybe neotest itself comes with some sort of configuration capabilities that I could use for this, to force each adapter to be limited to a certain filetype/pattern or language?
Any ideas?

For completeness, [this](https://github.com/fredrikaverpil/dotfiles/blob/main/nvim-lazyvim/lua/plugins/test.lua) is my neotest config.

**Comment by rcarriga** [comment, marked as answer]
I don't think it'd make sense to have this in neotest core. You can already configure adapters per project which will suit most people, and for shared projects the adapters should be able to detect filetypes etc. Having a third option for configuring adapters for a file would just be more complexity for very little payoff IMO.

If there is a significant advantage that I've missed then it could be worth it but as far as I can see there is little benefit

**Reply by fredrikaverpil** [reply]
Thanks for looking into this. I'll read up on per-project configs.

**Reply by fredrikaverpil** [reply]
I might add that I have a project which has both Python tests and Mocha tests. I feel there is a risk that the Mocha adapter will still perform an undesired "hostile takeover" of the Python tests with per-project configs... but I'll see what I can do.

**Reply by rcarriga** [reply]
It looks like the mocha adapter will match any file with `test` in the name https://github.com/adrigzr/neotest-mocha/blob/c22fc33015c4de4788b4a9394aeed6d7f4f1e949/lua/neotest-mocha/util.lua#L351-L353 which is odd, especially as it has more complex behaviour directly after that. I would suggest asking why the initial `test` match is there and if it can be removed/disabled

**Reply by fredrikaverpil** [reply]
For completeness, the above linked code has been removed from `neotest-mocha`, as this was the root cause of this issue. Thanks again @rcarriga !
